packages/react-workers/src/workers/viewportFeatures.worker.js

```diff
@@ -144,8 +148,34 @@ function getTimeSeries({ filters, column, stepSize, operation, operationColumn }
   postMessage({ result });
 }
 
-function getFilteredFeatures(filters) {
-  return !Object.keys(currentViewportFeatures).length
-    ? currentViewportFeatures
-    : currentViewportFeatures.filter(_buildFeatureFilter({ filters }));
+// See sorting details in utils/sorting.js
+function getRawFeatures({
+  filters,
+  limit = 10,
+  page = 1,
+  sortBy = [],
+  sortByDirection = 'asc'
+}) {
+  let data = [];
+  let numberPages = 0;
+
+  if (currentViewportFeatures) {
+    const features = applySorting(getFilteredFeatures(filters), {
+      sortBy,
+      sortByDirection
+    });
+
+    if (limit) {
+      numberPages = Math.ceil(features.length / limit);
```

Comment:
You should create an `applyLimit` function and simply call:

```js
data = features
if (limit) data = applyLimit(features, limit)

postMessage(......)
```

Reply:
I like it. I'll change it. In the future, maybe you can add that suggestion like a commit suggestion to just commit the change.

Reply:
Updated